server/routes/index.js

```diff
@@ -1,10 +1,10 @@
 const express = require('express');
 const addUserHandler = require('./addUser');
 const addCardHandler = require('./addCard');
-
+const {getProduct} = require('../controllers/index')
 const router = express.Router();
 
 router.post('/singup', addUserHandler);
 router.post('/addcard', addCardHandler);
-
+router.get('/api/product/:id',getProduct)
```

Comment:
I believe you called it `productId`
```suggestion
router.get('/api/product/:productId',getProduct)
```